Resolve bugs in Compare graphs

ak476519 opened the pull request:
This PR is a fix for the following issues: 

Fixes #461 where compare graph does not fetch data for groups, fixed the post request in compressed Readings API that's causing this problem.
Fixes #403 turns out to be a bug during Redux state change. When the selected groups/meters change, there's an incongruence between the time interval in Redux state(which is used to check if the desired data is available when graphing compare) and the time interval of data fetched from database by this request. Added code so that changes in meter/group selection affect the Redux state the same way changes in compare periods affect Redux states.

Commit:
test

Commit:
add code to reproduce graph through url,
with user selected intervals

Commit:
merge with changes in
OED/development branch

Commit:
remove json files to fix merging error

ak476519 committed:
Merge remote-tracking branch 'upstream/development' into development

ak476519 committed:
rewind package setting to honor changes in PR#486

ak476519 committed:
Merge remote-tracking branch 'upstream/development' into development

ak476519 committed:
fix bug in display of sorting drop-down menu

ak476519 committed:
remove residue of previous function

ak476519 committed:
fix bug: compare not able to fetch data as groups

ak476519 committed:
fix bug in compare graph display
(graphs do not update until a compare period tab is clicked)

ak476519 committed:
revert sort button back to using reactstrap
since the bug is fixed in another PR

huss commented on src/client/app/actions/compareReadings.ts:
```diff
@@ -80,13 +81,12 @@ function receiveGroupCompareReadings(groupIDs: number[], timeInterval: TimeInter
 /**
  * Fetch the data for the given meters over the given interval. Fully manages the Redux lifecycle.
  * @param {[number]} meterIDs The IDs of the meters whose data should be fetched
- * @param {TimeInterval} timeInterval The time interval over which data should be fetched
```
These two params seem to be replaced by comparePeriod and probably the same as param in next function.

huss commented on src/client/app/containers/ChartLinkContainer.ts:
```diff
@@ -20,6 +20,12 @@ function mapStateToProps(state: State) {
 			linkText += `&barStacking=${state.graph.barStacking}`;
 			break;
+		case 'line':
+			linkText += `&serverRange=${state.graph.timeInterval.toString()}`;
```
Should these commented out lines stay?

huss committed:
Merge branch 'development' of https://github.com/OpenEnergyDashboard/OED into pr/ak476519/501

huss committed:
fix linting issues

The changes to graph.ts are due to the merging with develoopment and
getting the same variable twice.

huss reviewed (changes requested):
The PR addresses the issues listed. It passes local testing. I have merged development and fixed linting issues. There are only a couple of comments about commenting that I hope can be addressed before the PR is merged.

huss committed:
Merge branch 'development' of https://github.com/OpenEnergyDashboard/OED into pr/ak476519/501

huss commented:
To move version 0.7 forward, the changes to comments are being delayed.
